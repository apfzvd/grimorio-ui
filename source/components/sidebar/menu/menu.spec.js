import '../../../internals/test/helper';

import Menu from './index';

/** @test {Menu} */
describe('Menu component', () => {

  const emptyProps = {
    children: undefined,
    className: undefined,
    items: undefined,
    type: undefined,
    theme: undefined,
  }

  let wrapper

  const items = [
    {
      key: 'dashboard',
      icon: 'dashboard',
      name: 'Dashboard',
      items: [
        { key: 'default', name: 'Default', link: '/default' },
        { key: 'ecommerce', name: 'eCommerce', link: '/ecommerce' },
        { key: 'newsportal', name: 'News Portal', link: '/news-portal' }
      ]
    },
    {
      key: 'charts',
      icon: 'insert_chart',
      name: 'Charts',
      items: [
        { key: 'test', name: 'Test', link: '/test' },
      ]
    }
  ]

  beforeAll(() => {
    wrapper = shallow(
      <Menu {...emptyProps} />
    );
  })

  describe('#render', () => {
    it('render with children', () => {
      wrapper.setProps({...emptyProps, children: <div>render com filho</div>})
      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('render with items', () => {
      wrapper.setProps({...emptyProps, items, type: 'accordionMenu', theme: 'dark', className: 'teste'})
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
