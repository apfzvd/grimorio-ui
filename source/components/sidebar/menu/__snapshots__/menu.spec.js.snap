// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component #render render with children 1`] = `
"<ul className=\\"menu default\\">
  <div>
    render com filho
  </div>
</ul>"
`;

exports[`Menu component #render render with items 1`] = `
"<ul className=\\"menu teste accordionMenu dark\\">
  <MenuItem icon=\\"dashboard\\" content=\\"Dashboard\\" link={[undefined]} items={{...}} active={[undefined]} />
  <MenuItem icon=\\"insert_chart\\" content=\\"Charts\\" link={[undefined]} items={{...}} active={[undefined]} />
</ul>"
`;
